src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1818,7 +1818,11 @@ private void HandleDirective(DirectiveDescriptor descriptor)
                             NextToken();
                             Balance(BalancingModes.NoErrorOnFailure, CSharpSymbolType.LeftBrace, CSharpSymbolType.RightBrace, startingBraceLocation);
                             Span.ChunkGenerator = new StatementChunkGenerator();
+                            var existingEditHandler = Span.EditHandler;
+                            Span.EditHandler = new CodeBlockEditHandler(Language.TokenizeString);
                             Output(SpanKindInternal.Code);
+
+                            Span.EditHandler = existingEditHandler;
```

Comment:
The jank-factor is high

Reply:
Damn straight lol, welcome to our parser.